test/Microsoft.AspNetCore.Server.KestrelTests/TestHelpers/MockSocket.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Threading;
+using Microsoft.AspNetCore.Server.Kestrel.Infrastructure;
+using Microsoft.AspNetCore.Server.Kestrel.Networking;
+
+namespace Microsoft.AspNetCore.Server.KestrelTests.TestHelpers
+{
+    class MockSocket : UvStreamHandle
+    {
+        public MockSocket(int threadId, IKestrelTrace logger) : base(logger)
+        {
+            var uv = new MockLibuv();
+            CreateMemory(uv, threadId, uv.handle_size(Libuv.HandleType.STREAM));
```

Comment:
Why did setting handle to 1 like before not work? `uv.handle_size` always returns `IntPtr.Size` BTW since we're using MockLibuv.